### ISSUE TITLE
VC++ still not supported?

### ISSUE THREAD

**Issue by jefffhaynes** [LucaFulchir/libRaptorQ#4]
While I realize your statement about VC++ not being a C++11 compiler used to be true, I'm not sure that's the case anymore.  

https://msdn.microsoft.com/en-us/library/hh567368.aspx

Although I'm having some trouble getting it to work it does seem like it's in the ballpark.  Any plans to give it a whirl?

**Comment by LucaFulchir** [comment]
Yeah, actually the compatibility is not that much of a problem anymore.
I'll fix it in the next couple of weeks.

Don't know if I'll work on vc++ compability on version 0.1 anymore, as it has some API problems,
and I'm changing a lot in 0.2, though. Version 0.1 was only intended to stabilize the algorithm, not really for production.

**Comment by jefffhaynes** [comment]
That's great, thanks!






On Thu, Apr 28, 2016 at 6:20 AM -0700, "Luca Fulchir" <notifications@github.com> wrote:












Yeah, actually the compatibility is not that much of a problem anymore.

I'll fix it in the next couple of weeks.



Don't know if I'll work on vc++ compability on version 0.1 anymore, as it has some API problems,

and I'm changing a lot in 0.2, though. Version 0.1 was only intended to stabilize the algorithm, not really for production.

**Comment by bloomsource** [comment]
cmake failed.

-- Configuring incomplete, errors occurred!
See also "/root/libRaptorQ-master/build/CMakeFiles/CMakeOutput.log".
See also "/root/libRaptorQ-master/build/CMakeFiles/CMakeError.log".


there is the tail of  /root/libRaptorQ-master/build/CMakeFiles/CMakeError.log
----------------------------------------------------------------------------------------------------------------
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /root/libRaptorQ-master/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_a589a/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_a589a.dir/build.make CMakeFiles/cmTC_a589a.dir/build
gmake[1]: Entering directory `/root/libRaptorQ-master/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_a589a.dir/CheckFunctionExists.c.o
/usr/bin/cc    -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTC_a589a.dir/CheckFunctionExists.c.o   -c /usr/local/share/cmake-3.5/Modules/CheckFunctionExists.c
Linking C executable cmTC_a589a
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_a589a.dir/link.txt --verbose=1
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTC_a589a.dir/CheckFunctionExists.c.o  -o cmTC_a589a -rdynamic -lpthreads
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_a589a] Error 1


centos 7.2 , cmake 3.5
i am very sure that pthread library is installed, 
i install development tool kit with 

yum groupinstall "Development Tools"

**Comment by LucaFulchir** [comment]
UPDATE:
A lot of time passed, but other things needed attention.

As of now, the status is "WONTFIX".  
Support for \<future\> seems to be broken as of MSVC 2015.

The only good news is that they seem to have already fixed it, but did not ship the new library as that would break compatibility.  

I will try to keep other things more compatible so that at least it compiles, even though it will not be usable until the next library release.

I will keep this ticket open until we can finally use it.

**Comment by supercairos** [comment]
Any news ? :)

**Comment by LucaFulchir** [comment]
uhm... yes and no.
The new version is tested under VS2017 and works, but I need some time before I can push the new commits on the public repo, as there was work done for a company in the middle.
